doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,156 @@
+# Audit Trail MVP
+
+NIH has required that DANDI Archive generate audit logs while leaving undefined
+what that means. We are choosing to fulfill the requirement with a feature that
+indicates to Dandiset owners and admins the sequence of significant actions
+taken to modify Dandisets, along with identification of who performed each
+action.
+
+The major motivation is to inform owners about how their Dandisets have changed
+when a given owner did not make the change and wishes to know who to speak to
+about those changes. In essence, it is a tool to support effective
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
```

Comment:
```suggestion
    - The ID of the Dandiset on which the action occurs (if performed on a dandiset endpoint or has dandiset as the parameter)
```

since could potentially be some operation on a blob or asset without

Reply:
The API doesn't appear to currently provide any mutating operations on blobs or assets without the context of a Dandiset.  I don't think it even could with the current architecture, as any change to an asset or blob produces a new asset with a new identifier, and in order for the operation to not be a waste of time, this new asset needs to be associated with one or more Dandisets.